### ISSUE TITLE
Incorrect parent/child self-relation behavior

### ISSUE THREAD

**Issue by canercandan** [graphcool/prisma#2071]
# Bug Report

Let's consider a type that contains two relations that allows for one to be related to the other and the otherwise around is true as well.

```gql
type Item {
  name: String! @unique
  parent: Item @relation(name: "ItemOnItem")
  children: [Item!]! @relation(name: "ItemOnItem")
}
```

I am creating those two `Item` with those mutations:

```gql
mutation {
  createItem( data: { name: "parent" } )
} {
  name
  parent { name }
  children { name }
}
```

```gql
mutation {
  createItem(
    data: {
       name: "child"
       parent: { connect: { name: "parent" } }
    }
  )
} {
  name
  parent { name }
  children { name }
}
```

**Current behavior**

```json
{
  "data": {
    "items": [
      {
        "name": "parent",
        "parent": null,
        "children": []
      },
      {
        "name": "child",
        "parent": {
          "name": "parent"
        },
        "children": [
          {
            "name": "parent"
          }
        ]
      },
    ]
  }
}
```

**Expected behavior?**

```json
{
  "data": {
    "items": [
      {
        "name": "parent",
        "parent": null,
        "children": [
          {
            "name": "child"
          }
        ]
      },
      {
        "name": "child",
        "parent": {
          "name": "parent"
        },
        "children": []
      },
    ]
  }
}
```

Anyone else can confirm this issue?

**Comment by do4gr** [comment]
Thanks for reporting this and the detailed description @canercandan 

The issue is now fixed on the unstable channel und should be released to the stable channel later today.

<img width="1264" alt="bildschirmfoto 2018-03-13 um 13 45 34" src="https://user-images.githubusercontent.com/16278407/37342487-ff38143e-26c4-11e8-997b-db7a3507a505.png">

**Comment by canercandan** [comment]
@do4gr amazing :raised_hands: thanks

**Comment by marktani** [comment]
This is now available in `1.4.0` 🙂
